nano/node/lmdb/lmdb.cpp

```diff
@@ -60,9 +60,15 @@ txn_tracking_enabled (txn_tracking_config_a.enable)
 
 		// Only open a write lock when upgrades are needed. This is because CLI commands
 		// open inactive nodes which can otherwise be locked here if there is a long write
-		// (can be a few minutes with the --fastbootstrap flag for instance)
+		// (can be a few minutes with the --fast_bootstrap flag for instance)
 		if (!is_fully_upgraded)
 		{
+			// Backup database before upgrade
+			if (backup_before_upgrade)
+			{
+				auto backup_path = path_a.parent_path () / "backup.upgrade.ldb";
+				mdb_env_copy (env, backup_path.string ().c_str ());
```

Comment:
If they wanted a backup we should probably inform them and exit the node if the backup fails (lack of disk space etc), so they can rectify it and try again.

Reply:
Good point